### ISSUE TITLE
Using MSELoss with different weights to replace KLDivloss

### ISSUE THREAD

**Issue by shuaigezhu** [shuaigezhu/star_galaxy_classification#29]

**Comment by shuaigezhu** [comment]
weights_label = float(temp_y[other_pixel].shape[0] / (temp_y[label_pixel].shape[0]+temp_y[other_pixel].shape[0]))
            weights_other = float(temp_y[label_pixel].shape[0] / (temp_y[label_pixel].shape[0]+temp_y[other_pixel].shape[0]))
            quasar_num = 2893
            star_num = 1669
            galaxy_num = 6771
            quasar_weight = (1/(1+quasar_num/star_num + quasar_num/galaxy_num))*weights_label 
            star_weight = (1/(1+star_num/quasar_num + star_num/galaxy_num))*weights_label
            galaxy_weight = (1/(1+galaxy_num/star_num + galaxy_num/quasar_num))*weights_label
            
            output_loss = weights_other*loss(logits[[other_pixel]], y[[other_pixel]])+quasar_weight*loss(logits[[label_pixel]][0], y[[label_pixel]][0])+star_weight*loss(logits[[label_pixel]][1], y[[label_pixel]][1])+galaxy_weight*loss(logits[[label_pixel]][2], y[[label_pixel]][2])

**Comment by kmyi** [comment]
I am not sure about the equation that you put there. Does this balance the three losses? Maybe you could plot all three losses after the weights are applied?

Kwang
--
Kwang Moo YI
Assistant Professor
Dept. of Computer Science
University of Victoria
+1 (250) 472 5837

**Comment by shuaigezhu** [comment]
![Screen Shot 2020-04-03 at 10 32 31 AM](https://user-images.githubusercontent.com/47727949/78388941-b9bdf680-7596-11ea-8520-d0468a752391.png)
![Screen Shot 2020-04-03 at 10 32 38 AM](https://user-images.githubusercontent.com/47727949/78388946-ba568d00-7596-11ea-9290-e5aac9044545.png)
![Screen Shot 2020-04-03 at 10 32 50 AM](https://user-images.githubusercontent.com/47727949/78388949-baef2380-7596-11ea-863d-4b5af7931f7b.png)
